src/cata_utility.cpp

```diff
@@ -18,9 +18,23 @@
 #include <string>
 #include <locale>
 
+double pow10( unsigned int n )
+{
+    double ret = 1;
+    double tmp = 10;
+    while( n ) {
+        if( n & 1 ) {
+            ret *= tmp;
+        }
+        tmp *= tmp;
+        n >>= 1;
+    }
+    return ret;
+}
+
 double round_up( double val, unsigned int dp )
 {
-    const double denominator = std::pow( 10.0, double( dp ) );
```

Comment:
Actually in my test std::pow( 10.0, 2 ) returned something like 100.00...0004, so that is not likely going to work.

Reply:
I think `std::round` will work, but I think it's better to use direct method instead of trying to correct wrong results. If it turns out to be neccessary to use `std::round` we can always change the definition of `pow10`. I do agree that we should add a comment explaining it though.